google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/BigQueryResultSet.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.bigquery;
+
+public interface BigQueryResultSet<T> {
+
+  /** Returns the schema of the results. */
+  Schema getSchema();
+
+  /**
+   * Returns the total number of rows in the complete result set, which can be more than the number
+   * of rows in the first page of results.
+   */
+  long getTotalRows();
```

Comment:
I'd consider not exposing this if we can get away with it.

Reply:
will follow up with Pavan on whether we need to expose it.

Reply:
Upon checking with Simba, we need to keep this field.